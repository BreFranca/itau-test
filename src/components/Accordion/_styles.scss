.btn_edit {
    margin-right: 15px;
}