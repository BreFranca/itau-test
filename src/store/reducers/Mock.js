export default [
    {
      id: 'c01bef7ef99b7c532501502f29d2b1a094ea14ea',
      title: 'Testando tudo',
      body: "<h3><strong>teste</strong></h3><h3><strong>das</strong></h3><h3><strong>as</strong></h3><h3><strong>d</strong></h3><h3><strong>asd</strong></h3><h3><strong>as</strong></h3><h3><strong>dasd</strong></h3>"
    },
    {
      id: 'd35ace893d64ddbeee92380558930344fb34c30c',
      title: 'TÃ­tulo 1',
      body: "<p>Teste 12313</p>"
    },
    {
      id: '23d8e11e2e54d88ac55dfbf60f6655c68cb45d70',
      title: 'Testando',
      body: ""
    },
    {
      id: 'a2208b748ce199d70d9dfe9f463f4e4abc62723f',
      title: 'Testando 123',
      body: "<p>teste</p>"
    },
    {
      id: '406e6e00863c1d03b67dd779c026461557a8f05d',
      title: 'Teste 1234',
      body: "<p>testando 31234</p><ul><li>dassa</li><li>d</li><li>sadasda</li></ul>"
    }
]
