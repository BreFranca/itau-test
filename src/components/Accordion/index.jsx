import React from 'react'
import {
    Collapse,
    Icon
} from 'antd'
import './_styles.scss'

const Accordion = ({ options, onEdit, onDelete }) => {
    const { Panel } = Collapse
    return (
        <Collapse accordion>
            {options && options.map((option, index) => (
                <Panel
                    header={option.title}
                    key={index}
                    extra={
                        <React.Fragment>
                            <Icon
                                type="edit"
                                onClick={onEdit}
                                className="btn_edit"
                            />
                            <Icon
                                type="delete"
                                onClick={onDelete}
                            />
                        </React.Fragment>
                    }
                >
                    <p>{option.text}</p>
                </Panel>
            ))}
        </Collapse>
    )
}

export default Accordion
