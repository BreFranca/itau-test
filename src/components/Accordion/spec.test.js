import React from 'react'
import Accordion from '.'
import Mock from './Mock'

import '../../enzyme'

import { mount } from 'enzyme'

describe('Accordion', () => {
    it('should be a instance of Accordion', () => {
        const mockFunc = jest.fn()
        const component = mount(
            <Accordion
                options={Mock}
                onEdit={mockFunc}
                onDelete={mockFunc}
            />
        )

        expect(component).toMatchSnapshot();

        expect(component.props().options).toEqual(Mock)

        component.unmount()
    })
})