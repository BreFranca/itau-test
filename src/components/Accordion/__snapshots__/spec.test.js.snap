// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should be a instance of Accordion 1`] = `ReactWrapper {}`;

exports[`Accordion should be a instance of Accordion 1`] = `ReactWrapper {}`;
